src/Metadata/Extractor/XmlResourceExtractor.php

```diff
@@ -0,0 +1,412 @@
+            'paginationUseOutputWalkers' => $this->phpize($resource, 'paginationUseOutputWalkers', 'bool'),
+            'paginationItemsPerPage' => $this->phpize($resource, 'paginationItemsPerPage', 'integer'),
+            'paginationMaximumItemsPerPage' => $this->phpize($resource, 'paginationMaximumItemsPerPage', 'integer'),
+            'paginationPartial' => $this->phpize($resource, 'paginationPartial', 'bool'),
+            'paginationType' => $this->phpize($resource, 'paginationType', 'string'),
+            'security' => $this->phpize($resource, 'security', 'string'),
+            'securityMessage' => $this->phpize($resource, 'securityMessage', 'string'),
+            'securityPostDenormalize' => $this->phpize($resource, 'securityPostDenormalize', 'string'),
+            'securityPostDenormalizeMessage' => $this->phpize($resource, 'securityPostDenormalizeMessage', 'string'),
+            'securityPostValidation' => $this->phpize($resource, 'securityPostValidation', 'string'),
+            'securityPostValidationMessage' => $this->phpize($resource, 'securityPostValidationMessage', 'string'),
+            'normalizationContext' => isset($resource->normalizationContext->values) ? $this->buildValues($resource->normalizationContext->values) : null,
+            'denormalizationContext' => isset($resource->denormalizationContext->values) ? $this->buildValues($resource->denormalizationContext->values) : null,
+            'validationContext' => isset($resource->validationContext->values) ? $this->buildValues($resource->validationContext->values) : null,
+            'filters' => $this->buildArrayValue($resource, 'filter'),
+            'order' => isset($resource->order->values) ? $this->buildValues($resource->order->values) : null,
+            'extraProperties' => $this->buildExtraProperties($resource, 'extraProperties'),
+        ];
+    }
+
+    private function buildFormats(\SimpleXMLElement $resource, string $key): ?array
+    {
+        if (!isset($resource->{$key}->format)) {
+            return null;
+        }
+
+        $data = [];
+        foreach ($resource->{$key}->format as $format) {
+            if (isset($format['name'])) {
+                $data[(string) $format['name']] = (string) $format;
+                continue;
+            }
+
+            $data[] = (string) $format;
+        }
+
+        return $data;
+    }
+
+    private function buildUriVariables(\SimpleXMLElement $resource): ?array
+    {
+        if (!isset($resource->uriVariables->uriVariable)) {
+            return null;
+        }
+
+        $uriVariables = [];
+        foreach ($resource->uriVariables->uriVariable as $data) {
+            $parameterName = (string) $data['parameterName'];
+            if (1 === \count($data->attributes())) {
+                $uriVariables[$parameterName] = $parameterName;
+                continue;
+            }
+
+            if (null !== ($property = $this->phpize($data, 'fromProperty', 'string'))) {
```

Comment:
does this needs to be `null !==` or can it be `!($property = $this->phpize($data, 'fromProperty', 'string'))` ?